### DISCUSSION TITLE
How to call re-centering

### DISCUSSION THREAD

**Discussion by md-hassan** [google/model-viewer#4164] [Q&A]
My function changes the object's orientation, sometimes leading to awkward zoom in/out. How do I call the re-centering function after I do the orientation change?
I am aware that tapping outside does re-centering, but I could not find a way to call 'tapping' as part of a function.

Thanks

**Comment by elalish** [comment]
If by re-centering you mean the `camera-target`, just use `getCameraTarget()` at the beginning and then reapply it to `camera-target` as desired.

**Reply by md-hassan** [reply]
![screenshot](https://user-images.githubusercontent.com/62975633/225517312-1d4f3ddc-42cb-41b8-92bd-1e0e9d6bc129.png)

Hi, thanks for the reply.

As you can see in the screenshot, the object is zoomed in awkwardly when I update the orientation. I want to set the camera so that the object is visible again (me calling it re-centering above might have caused confusion). In effect, I want to do what tapping anywhere outside the model does. So, finally, the object should look like this (after I tap anywhere outside the model):

![image](https://user-images.githubusercontent.com/62975633/225518195-05efcde6-a461-4002-aacd-eca50bd6600c.png)

As for the suggestion to use the `camera-target`, changing `orientation` does not affect the `camera-target` (see screenshot 1). Not sure how how that can be used here.
 
Thanks!
